service/javadsl/server/src/main/java/com/lightbend/lagom/javadsl/server/ServiceGuiceSupport.java

```diff
@@ -30,6 +33,86 @@
  */
 public interface ServiceGuiceSupport extends ServiceClientGuiceSupport {
 
+
+    // TODO: document usage and explain when and how to prefix
+    @ApiMayChange
+    default <R extends Router> AdditionalRouter additionalRouter(Class<R> router) {
```

Comment:
Here I'm using `play.routing.Router` which is part of the Java API in Play, but I guess we need to replace it by the Scala one, `play.api.routing.Router`. 

The reason is that the router generator in Play will generate `play.api.routing.Router` even for Java and later the Scala router is wrapped on a `JavaRouterAdapter`. For a user that will be easier if they can use the Scala class instead. 

However, we still need to think if this is also a good choice for gRPC or Routers derived from `akka-http` `Routes`. 

@raboof / @marcospereira, WDYT?

Reply:
It seems kind of weird to use the play scaladsl `play.api.routing.Router` in the Lagom javadsl.

On the other hand, indeed the Play-gRPC code generation always generates scaladsl `play.api.routing.Router` glue, so it *is* the most convenient at this stage.

I would be in favour of adding the `additionalRouter` with the scaladsl Router class. This should give us what we need for gRPC, and if at a later stage we also want to support javadsl Routers in this class we can add those with a different method name.

Reply:
yep, I will change it back to the scaladsl version. It's strange indeed, but in both cases (gRPC and Play's `routes` file), the generated version is a `play.api.routing.Router`. 

The user just gets a `Router.class` to bind. The fact that it was generated from a Scala file is a detail. At least when binding it, not when debugging.

Reply:
I just recalled that @marcospereira mentioned the existence of a `RoutingDsl` that allows users to build a Java Router programmatically. 

Nonetheless, I still think that we should move to `play.api.routring.Router` for the reasons we mentioned already. 

I see the `RoutingDsl` as a border case. NB, this is only a problem for users using Lagom and willing to add a programmatically built Java Router. I would leave this option on the side for now.

Let's move to the Scala DSL router.

Reply:
Agreed completely! I think users will rarely want to add hand-built custom `Router`s to an application, more likely controllers (such as the file upload one). But even that I'd consider outside of the scope of this PR. :+1: